### ISSUE TITLE
9.3.16 Windows  version has problems 

### ISSUE THREAD

**Issue by xpxaxsxi** [SWI-Prolog/swipl-devel#1338]
My machine is Windows 11 Pro and the swipl is 64-bits. 
UPDATE  1) I have startup scripts that could cause this, 2) I have problems in application that uses pengines
  
```prolog
1 ?- version.
Welcome to SWI-Prolog (threaded, 64 bits, version 9.3.16)
SWI-Prolog comes with ABSOLUTELY NO WARRANTY. This is free software.
Please run ?- license. for legal details.

For online help and background, visit https://www.swi-prolog.org
For built-in help, use ?- help(Topic). or ?- apropos(Word).

true.

2 ?- debug(_).
true.

3 ?- make.
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/engines.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/bags.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/bags.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/engines.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/bags.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/engines.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/bags.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/engines.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
Warning: The predicates below are not defined. If these are defined
Warning: at runtime using assert/1, use :- dynamic Name/Arity.
Warning: 
Warning: '$bags':'$call_cleanup'/0, which is referenced by
Warning:        z:/home/swipl/src/swipl-devel/build.win64/home/boot/bags.pl:104: 1-st clause of '$bags':cleanup_bag/2
Warning: qsave:qsave_toplevel/0, which is referenced by
Warning:        z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl:914: 1-st clause of '$toplevel':'$compile_'/0
true.
```

Without startup scripts I get pretty much the same errors I think in a cmd line swipl
```prolog
C:\Users\pasik>"c:\Program Files\swipl\bin\swipl.exe" -f none
Welcome to SWI-Prolog (threaded, 64 bits, version 9.3.16)
SWI-Prolog comes with ABSOLUTELY NO WARRANTY. This is free software.
Please run ?- license. for legal details.

For online help and background, visit https://www.swi-prolog.org
For built-in help, use ?- help(Topic). or ?- apropos(Word).

1 ?- debug(_):
ERROR: Stream user_input:7:15 Syntax error: Unexpected end of file
Action (h for help) ? abort
% Execution Aborted
1 ?- debug(_).
true.

2 ?- make.
% Got exception existence_error(procedure,prolog:history/2) (Ctx0=system:catch/3, Catcher='$history':'$save_history_line'/1)
% Got exception existence_error(source_sink,'z:/home/swipl/src/swipl-devel/build.win64/home/boot/toplevel.pl') (Ctx0=system:open/4, Catcher=prolog_clause:read_term_at_line_2/6)
```

**Comment by JanWielemaker** [comment]
This was also reported on Discourse.   Fixed in 9.3.17, which has been uploaded a few minutes ago.  Should be public within an hour.

**Comment by xpxaxsxi** [comment]
Hey can I ask the percentage of Windows users of swi-prolog, your rough estimate is fine
